inject/src/main/java/io/avaje/inject/BeanScope.java

```diff
@@ -140,7 +141,51 @@ static BeanScopeBuilder builder() {
   <T> T get(Class<T> type, @Nullable String name);
 
   /**
-   * Return a single bean given the generic type and name.
+   * Return a single bean given its type and generic components.
+   *
+   * <pre>{@code
+   * Map<String, CoffeeMaker> coffeeMaker = beanScope.get(Map.class, String.class, CoffeeMaker.class);
```

Comment:
This looks like extra functionality and not part of the bug fix?

Improve this example